frontend/catalyst/jax_primitives.py

```diff
@@ -1476,3 +1474,10 @@ def _qfor_lowering(
 mlir.register_lowering(func_p, _func_lowering)
 mlir.register_lowering(jvp_p, _jvp_lowering)
 mlir.register_lowering(vjp_p, _vjp_lowering)
+
+# pylint: disable=protected-access
+api_util._shaped_abstractify_handlers[type] = lambda x: core.ShapedArray([], x)
```

Comment:
Oops I think I forgot to push a bunch changes before handing it over, for example here I had the following which might be a bit safer and also creates a weak type:
```py
def _scalar_abstractify(t):
    if isinstance(t, (int, float, complex)):
        raise TypeError(f"Cannot convert given type {t} scalar ShapedArray.")
    return jax.ShapedArray([], dtype=t, weak_type=True)
_shaped_abstractify_handlers[type] = _scalar_abstractify
```

Reply:
https://github.com/PennyLaneAI/catalyst/pull/143/commits/e63b067523b92e09239a8f0ff2c769a5c1dd4b41